src/zc.adoc

```diff
@@ -0,0 +1,2611 @@
+    { bits:  3, name: 0x4, attr: ['FUNCT3'] },
+],config:{bits:16}}
+....
+
+Description:
+
+This instruction takes a single source/destination operand. 
+It zero-extends the least-significant word of the operand to XLEN bits by inserting zeros into all of
+the bits more significant than 31.
+
+[NOTE]
+====
+_rd'/rs1'_ is from the standard 8-register set x8-x15.
+====
+
+Prerequisites:
+
+Zba is also required.
+ 
+32-bit equivalent:
+
+[source,sail]
+----
+add.uw rd'/rs1', rd'/rs1', zero
+----
+
+[NOTE]
+====
+The SAIL module variable for _rd'/rs1'_ is called _rsdc_.
+====
+
+Operation:
+
+[source,sail]
+----
+X(rsdc) = EXTZ(X(rsdc)[31..0]);
+----
+
+<<<
+[#insns-c_not,reftext="Bitwise not, 16-bit encoding"]
+==== c.not
+
+Synopsis:
+
+Bitwise not, 16-bit encoding
+
+Mnemonic:
+
+c.not _rd'/rs1'_
+
+Encoding (RV32, RV64):
+
+[wavedrom, , svg]
+....
+{reg:[
+    { bits:  2, name: 0x1, attr: ['C1'] },
+    { bits:  3, name: 0x5, attr: ['C.NOT'] },
+    { bits:  2, name: 0x3, attr: ['FUNCT2'] },
+    { bits:  3, name: 'rd\'/rs1\'', attr: ['SRCDST'] },
+    { bits:  3, name: 0x7 },
+    { bits:  3, name: 0x4, attr: ['FUNCT3'] },
+],config:{bits:16}}
+....
+
+Description:
+
+This instruction takes the one's complement of _rd'/rs1'_ and writes the result to the same register.
+
+[NOTE]
+====
+  _rd'/rs1'_ is from the standard 8-register set x8-x15.
+====
+
+Prerequisites:
+
+None
+
+32-bit equivalent:
+
+[source,sail]
+----
+xori rd'/rs1', rd'/rs1', -1
+----
+
+[NOTE]
+====
+The SAIL module variable for _rd'/rs1'_ is called _rsdc_.
+====
+
+Operation:
+
+[source,sail]
+----
+X(rsdc) = X(rsdc) XOR -1;
+----
+
+<<<
+[#insns-c_mul,reftext="Multiply, 16-bit encoding"]
+==== c.mul
+
+Synopsis:
+
+Multiply, 16-bit encoding
+
+Mnemonic:
+
+c.mul _rsd'_, _rs2'_
+
+Encoding (RV32, RV64):
+
+[wavedrom, , svg]
+....
+{reg:[
+    { bits:  2, name: 0x1, attr: ['C1'] },
+    { bits:  3, name: 'rs2\'', attr: ['SRC2']  },
+    { bits:  2, name: 0x2, attr: ['FUNCT2'] },
+    { bits:  3, name: 'rd\'/rs1\'', attr: ['SRCDST'] },
+    { bits:  3, name: 0x7 },
+    { bits:  3, name: 0x4, attr: ['FUNCT3'] },
+],config:{bits:16}}
+....
+
+Description:
+
+This instruction multiplies XLEN bits of the source operands from _rsd'_ and _rs2'_ and writes the lowest XLEN bits of the result to _rsd'_.
+
+[NOTE]
+====
+_rd'/rs1'_ and _rs2'_ are from the standard 8-register set x8-x15.
+====
+
+Prerequisites:
+
+M or Zmmul must be configured.
+//
+//32-bit equivalent:
+//
+//<<insns-mul>>
+
+[NOTE]
+====
+The SAIL module variable for _rd'/rs1'_ is called _rsdc_, and for _rs2'_ is called _rs2c_.
+====
+
+Operation:
+
+[source,sail]
+----
+let result_wide = to_bits(2 * sizeof(xlen), signed(X(rsdc)) * signed(X(rs2c)));
+X(rsdc) = result_wide[(sizeof(xlen) - 1) .. 0];
+----
+
+<<<
+
+[#insns-pushpop,reftext="PUSH/POP Register Instructions"]
+=== PUSH/POP register instructions
+
+These instructions are collectively referred to as PUSH/POP: 
+
+* <<#insns-cm_push>> 
```

Comment:
the reference text here should probably be the instruciton name, this list reads strangely

Reply:
That was from the original spec, but I changed it to the instruction names as per your suggestion.